core/client/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -274,24 +260,30 @@ private boolean tooManyPacketsInFlight() {
   }
 
   /**
-   * Sends an empty packet to signify the EOF.
+   * Sends an EOF or a CANCEL packet to end the write request if the stream.
+   *
+   * @param isEof whether it is an EOF request or a CANCEL request
    */
-  private void sendEOF() {
+  private void sendEOFOrCancel(boolean isEof) {
```

Comment:
@gpang Are you ok with separating this since you preferred to reuse? I can revert this to the earlier version that has two separate functions.  I am indifferent. There is some code reuse but mostly boilerplate (lock, unlock).

Similar reply for other similar comments.